libr/cons/cons.c

```diff
@@ -13,6 +11,39 @@ static RCons s_cons_global = {0};
 
 static void __break_signal(int sig);
 
+// Count display lines, treating long lines (>80 chars) as multiple lines
+// Stops counting at R_CONS_MAX_LINES to avoid performance issues with huge outputs
+#define R_CONS_MAX_LINES 10000
+
+static unsigned int count_display_lines(const char *buffer, size_t len) {
+	if (!buffer || len == 0) {
+		return 0;
+	}
+	unsigned int lines = 0;
+	unsigned int line_chars = 0;
+	size_t i;
+	for (i = 0; i < len && lines < R_CONS_MAX_LINES; i++) {
+		if (buffer[i] == '\n') {
+			lines++;
+			line_chars = 0;
+		} else {
+			line_chars++;
+			// Count extra line for every 80 characters in a single line
+			if (line_chars > 0 && line_chars % 80 == 0) {
```

Comment:
let's remove the hardcoded "80" value here. use const int columns = r_cons_get_size